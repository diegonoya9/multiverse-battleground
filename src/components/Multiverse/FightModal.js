// En tu componente
import React from 'react';
import classes from './FightModal.module.css'; // AsegÃºrate de importar tu archivo de estilos correctamente

const FightModal = ({ onStartAdventure }) => {
    return (
        <div className={classes.modalWrapper}>
            <div className={classes.modalContainer}>
                <div className={classes.modalContent}>
                    <h1 style={{ color: '#ff4500' }}>Welcome to the Multiverse Battleground</h1>
                    <p style={{ color: '#fff' }}>Get ready for epic battles!</p>
                    <button className={classes.modalButton} onClick={onStartAdventure}>
                        Start Adventure!
                    </button>
                </div>
            </div>
        </div>
    );
};

export default FightModal;
