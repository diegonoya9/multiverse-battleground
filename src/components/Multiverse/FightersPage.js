import classes from "./FightersPage.module.css"
import ReactAudioPlayer from 'react-audio-player';
import Modal from "../UI/Modal";
import { useState, useContext, useEffect, useCallback } from "react";
import musicFile from "../../assets/sounds/music/DirtyLove.WAV"
import Button from "../UI/Button";
import { MyContext } from "../../context/MyContext";
import FighterCard from "../UI/FighterCard";
import { useTranslation } from 'react-i18next';
const FightersPage = ({ user, changeMultiverseActivePage }) => {
    const { t } = useTranslation();
    const priceTags = document.querySelectorAll(".fighterPriceBlock")
    priceTags.forEach(tag => {
        tag.style.display = "none"
    });;
    const [showModal, setShowModal] = useState(false)
    const [modalContent, setModalContent] = useState()
    const { userContext } = useContext(MyContext);
    const [fighters, setFighters] = useState()
    const [moves, setMoves] = useState()
    /*const [showConfirm, setShowConfirm] = useState(false)
    const [userFighterId, setUserFighterId] = useState(false)*/
    const [allowCloseModal, setAllowCloseModal] = useState(true)
    let backEndUrl = userContext.backEndUrl
    let activeUser = userContext.idUsuario
    const audioStyle = {
        display: 'none',
    };
    const closeModal = () => {
        if (allowCloseModal) {
            setShowModal(false)
            setModalContent()
        }
    }
    const addToParty = (userFighterId) => {
        let cant = 0
        let cantExceeded = false
        /* Checks if there are already 4 fighters in the party */
        fighters.forEach((fighter) => {
            if (fighter.in_party === "true") {
                cant++
            }
            if (cant === 4) {
                cantExceeded = true
            }
        })
        if (cantExceeded) {
            setAllowCloseModal(true)
            setModalContent("Can't add more than 4 members to the party.")
            setShowModal(true)
        } else {
            const parameters = [{
                user_fighter_id: userFighterId
            }]
            setModalContent('Adding to party...')
            setAllowCloseModal(false)
            setShowModal(true)
            fetch(backEndUrl + "/addtoparty", {
                method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(parameters),
            }).then(() => {
                updateFighters()
            })
        }
    }
    const removeFromParty = (user_fighter_id) => {
        let newUser = user
        const parameters = [{
            user_fighter_id,
            user_id: newUser.user_id
        }]
        setModalContent('Removing from party...')
        setAllowCloseModal(false)
        setShowModal(true)
        fetch(backEndUrl + "/removefromparty", {
            method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(parameters),
        }).then(() => updateFighters())

    }
    const viewActions = (move_id) => {
        let actions = []
        moves.forEach((move) => {
            if (move.move_id === move_id) {
                actions.push(move.actionmoves)
            }
        })
        setAllowCloseModal(true)
        if (actions[0]) {
            setModalContent(actions[0].map((action, index) => {
                return <div key={Math.random()} >
                    <p>Action: {index + 1}</p>
                    <p>Inflicted on: {action.inflicted_on}</p>
                    <p>Type: {action.attack_type}</p>
                    <p>Field: {action.field}</p>
                    <p>Value: {action.value}</p>
                </div>
            }))
            setShowModal(true)
        }
    }
    useEffect(() => {
        if (moves) {
            setModalContent(
                <ul>{moves.map((move) => {
                    return <Button key={move.name} onClick={() => { viewActions(move.move_id) }}>
                        {move.name}
                    </Button>
                })} </ul>
            )
        }
    }, [moves])
    const viewMovements = (user_fighter_id) => {
        let moves = []
        fighters.forEach((fighter) => {
            if (fighter.user_fighter_id === user_fighter_id) {
                fighter.moves.forEach((move) => {
                    moves.push(move.moves)
                })
            }
        })
        setAllowCloseModal(true)
        setMoves(moves)
    }
    useEffect(() => {
        if (modalContent) {
            setShowModal(true)
        }
    }, [modalContent])
    const setFirstFighter = (user_fighter_id) => {
        const parameters = [{
            user_id: user.user_id,
            user_fighter_id
        }]
        setModalContent('Assigning first fighter')
        setShowModal(true)
        setAllowCloseModal(false)
        fetch(backEndUrl + "/setfirstfighter", {
            method: 'POST', // O 'PUT' si deseas sobrescribir completamente los datos del usuario
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(parameters),
        }).then(() => {
            updateFighters()
        })
    }
    const deleteFighter = (userFighterId) => {
       /* setUserFighterId(userFighterId)
        setShowConfirm(true)
        setShowModal(true)*/
    }
    const deleteUserFighter = async (userFighterId) => {
        /* let deleteFighter = await user.fighters.filter((fighter) => {
             return fighter.userFighterId === userFighterId
         })
         let newFighters = user.fighters.filter((fighter) => {
             return fighter.userFighterId !== userFighterId
         })
         const sell = user.objects[4].quantity
         let price = deleteFighter[0].price
         const result = sell + price
         let newUser = user
         newUser.objects[4].quantity = result
         newUser.fighters = newFighters
         setShowModal(false)
         setShowConfirm(false)*/
    }
    const fetchData = async () => {
        try {
            if (backEndUrl && activeUser) {
                const response = await fetch(backEndUrl + '/alluserfighters/' + activeUser);
                const data = await response.json();
                setFighters(data);
            }
        } catch (error) {
            console.error('Error fetching data:', error);
        }
    };
    const updateFighters = async () => {
        await fetchData()
    }

    useEffect(() => {
        setShowModal(false)
        setModalContent()
    }, [fighters])
    useEffect(() => {
        updateFighters()
    }, [])
    return (<div alt="divContainerFightersPage" className={`${classes.body} ${classes.backgroundImg}`}>
        <Button colorType="lightgreen" value={t('fighterspage.back')} onClick={() => { changeMultiverseActivePage("mainMenu") }}></Button>
        <div className={classes.container} >
            <ReactAudioPlayer src={musicFile} autoPlay controls style={audioStyle} />
            {fighters &&
                fighters.map((fighter, i) => {
                    return (<div className={classes.fighterContainer} key={fighter.user_fighter_id}>
                        <FighterCard fighter={fighter}></FighterCard>
                        <Button onClick={() => { if (fighter.in_party === "true") { setFirstFighter(fighter.user_fighter_id) } else { setAllowCloseModal(true); setModalContent('You need to add the fighter to the party first.') } }} value={t('fighterspage.setFirst')}></Button>
                        <Button onClick={() => { viewMovements(fighter.user_fighter_id) }} value={t('fighterspage.viewMovements')}></Button>
                        {fighter.in_party === "true" ?
                            <Button onClick={() => { removeFromParty(fighter.user_fighter_id) }} value={t('fighterspage.removeFromParty')}></Button>
                            :
                            <Button onClick={() => { addToParty(fighter.user_fighter_id) }} value={t('fighterspage.addToParty')}></Button>
                        }
                        {/* <Button onClick={() => { deleteFighter(fighter.user_fighter_id) }} value="Sell Fighter"></Button> */}
                    </div>
                    );
                })}
        </div>
        {/*showModal && !modalContent && <Modal styleType={"battlegroundColiseum"} onClose={closeModal} color="white">
            {showConfirm && <div>
                <h3>Are you sure you want to sell this fighter?</h3>
                <Button onClick={() => deleteUserFighter(userFighterId)}>Sell</Button>
            </div>}
            </Modal>*/}
        {showModal && modalContent && <Modal styleType={"battlegroundColiseum"} onClose={closeModal} color="white">
            {modalContent}
        </Modal>}
    </div >
    );

}

export default FightersPage