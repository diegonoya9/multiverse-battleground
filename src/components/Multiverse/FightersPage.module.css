* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

/* .body {
    background: #282c34;
} */
.backgroundImg {
    height: auto;
    min-height: 100vh;
    width: 100vw;
    background-image: url('../../assets/img/island.gif');
    background-repeat: no-repeat;
    background-size: 100% 100%;
    background-attachment: fixed;

}

.container {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: flex-start;
    min-height: 100vh;
    /* background: #282c34; */
    gap: 1em;
    margin-left: 25px;
}

.fighterContainer {
    display: flex;
    flex-direction: column;
    align-items: stretch;
}

.title {
    text-transform: capitalize;
}