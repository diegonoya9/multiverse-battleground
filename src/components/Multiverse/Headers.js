const Headers = () => {
    return (
        <div>
            <link rel="preconnect" href="https://fonts.googleapis.com" />
            <link rel="preconnect" href="https://fonts.gstatic.com" />
            <link href="https://fonts.googleapis.com/css2?family=Allerta+Stencil&family=Merienda&family=Mochiy+Pop+One&family=Dhurjati&family=Black+Ops+One&family=Orbitron&display=swap" rel="stylesheet" />
        </div>
    )
}
export default Headers