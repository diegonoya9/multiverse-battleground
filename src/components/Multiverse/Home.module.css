* {
    font-family: 'Silkscreen', sans-serif;
}

.homeDiv {
    height: 100vh;
    width: 100vw;

}


.modalButton {
    background: #ff4500;
    color: #fff;
    padding: 10px 20px;
    font-size: 18px;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    margin-top: 20px;
}