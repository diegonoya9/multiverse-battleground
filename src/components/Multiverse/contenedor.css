* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: sans-serif;
}

.body {
    background: #282c34;
}

.container {
    margin-top: 2em;
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    min-height: 100vh;
    background: #282c34;
    gap: 3em;
}

.card {
    position: relative;
    width: 300px;
    height: 350px;
}

.card .face {
    position: absolute;
    width: 100%;
    height: 100%;
    backface-visibility: hidden;
    border-radius: 10px;
    overflow: hidden;
    transition: .5s;
}

.card .front {
    transform: perspective(600px) rotateY(0deg);
    box-shadow: 0 5px 10px #000;
}

.title {
    text-transform: capitalize;
}

.card .front img {
    position: absolute;
    width: 40%;
    height: 50%;
    object-fit: cover;
}

.card .front h3 {
    position: absolute;
    bottom: 0;
    width: 100%;
    padding: 10px;
    height: 45px;
    color: #fff;
    background: rgba(0, 0, 0, .4);
    text-align: center;
}

.card .back {
    transform: perspective(600px) rotateY(180deg);
    background-color: #1b1d20;
    padding: 15px;
    color: #fff;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    text-align: center;
    box-shadow: 0 5px 10px #000;
}

.card .back button {
    border-top: solid 1px #f3f3f3;
    height: 25px;
    line-height: 25px;
}

.card .back p {
    font-size: 14px;
    margin-top: 5px;
    letter-spacing: 2px;
}

.card:hover .front {
    transform: perspective(600px) rotateY(180deg);
}

.card:hover .back {
    transform: perspective(600px) rotateY(360deg);
}