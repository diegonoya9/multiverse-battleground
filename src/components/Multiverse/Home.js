import { useState } from "react"
import Multiverse from "./Multiverse"
import FightModal from "./FightModal"
import Battleground from "../Fights/Battleground"
import classes from './Home.module.css'

const Home = () => {
    const [showModal, setShowModal] = useState(true);

    const handleStartAdventure = () => {
        console.log(showModal)
        // Aquí puedes añadir lógica para iniciar la aventura
        console.log('Adventure started!');
        setShowModal(false);
    };
    const [activePage, setActivePage] = useState(() => {
        if (process.env.NODE_ENV === 'production') {
            // Código específico para el entorno de desarrollo
            return 1
        } else if (process.env.NODE_ENV === 'development') {
            // Código específico para el entorno de producción
            return 1
        }
    })
    const changeActivePage = (page) => {
        setActivePage(page)
    }
    return (
        <div id="homeDiv" className={classes.homeDiv}>
            {showModal && <FightModal showModal={showModal} onStartAdventure={handleStartAdventure} />}
            {activePage && activePage === 1 && !showModal && <Multiverse changeActivePage={changeActivePage}></Multiverse>}
            {activePage && activePage === 2 && !showModal && <Battleground changeActivePage={changeActivePage}></Battleground>}
        </div>
    )
}

export default Home