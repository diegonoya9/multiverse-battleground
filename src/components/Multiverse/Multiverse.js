import { memo, useState, useEffect, useContext } from "react"
import FightersPage from "./FightersPage.js"
import ObjectsPage from "./ObjectsPage.js"
import UsersPage from './UsersPage.js'
import classes from './Multiverse.module.css'
import ShopPage from "./ShopPage.js"
import ReactAudioPlayer from 'react-audio-player';
import musicFile from "../../assets/sounds/music/Aeroplane.WAV"
import Button from "../UI/Button.js"
import { MyContext } from '../../context/MyContext';

const Multiverse = ({ changeActivePage }) => {
    const audioStyle = {
        display: 'none', // Oculta el reproductor de audio visualmente
    };
    const { userContext, setUserId } = useContext(MyContext);
    let activeUser = userContext.idUsuario
    const [multiverseActivePage, setMultiverseActivePage] = useState("mainMenu")
    const changeMultiverseActivePage = (activePage) => {
        setMultiverseActivePage(activePage)
    }
    const [user, setUser] = useState()
    const [money, setMoney] = useState()
    useEffect(() => {
        if (user) {
            let newMoney = user.objects.filter((object) => {
                return object.name === "money"
            })
            setMoney(newMoney[0].quantity)
        }
    }, [user])
    const updateUser = () => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
    }
    useEffect(() => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
        const audio = document.getElementById('audioPlayer');
        if (audio) {
            audio.play()
        }
    }, [])
    useEffect(() => {
        fetch('https://multiverse-battleground-default-rtdb.firebaseio.com/users/' + activeUser + '.json')
            .then(response => response.json())
            .then(data => { setUser(data) })
    }, [multiverseActivePage])
    return (
        <div className={classes.container}>
            {multiverseActivePage === 'mainMenu' && user && (
                <div className={`${classes.mainMenu} ${classes.container}`}>
                    <ReactAudioPlayer src={musicFile} autoPlay id="audioPlayer" controls style={audioStyle} />
                    <h1 className={classes.h1}>Welcome {user.name}</h1>
                    <h2 className={classes.h2}>You have {money} pesos</h2>
                    <Button value='Fight' colorType="red" onClick={() => changeActivePage(2)}></Button>
                    <Button value='Fighters' colorType="brown" onClick={(e) => {
                        changeMultiverseActivePage('fighters');
                    }}></Button>
                    <Button value='Bag' colorType="green" onClick={(e) => {
                        changeMultiverseActivePage('bag');
                    }}></Button>
                    <Button value='Shop' colorType="yellow" onClick={(e) => {
                        changeMultiverseActivePage('shop');
                    }}></Button>
                    <Button value='Users' colorType="blue" onClick={(e) => {
                        changeMultiverseActivePage('users');
                    }}></Button>
                </div>
            )}
            {multiverseActivePage === "fighters" && <FightersPage
                updateUser={updateUser} changeMultiverseActivePage={changeMultiverseActivePage} user={user}></FightersPage>}
            {multiverseActivePage === "bag" && <ObjectsPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></ObjectsPage>}
            {multiverseActivePage === "shop" && <ShopPage changeMultiverseActivePage={changeMultiverseActivePage} ></ShopPage>}
            {multiverseActivePage === "users" && <UsersPage changeMultiverseActivePage={changeMultiverseActivePage} ></UsersPage>}
        </div>

        /*<div>
            {multiverseActivePage === "mainMenu" && user &&
                <div>
                    <ReactAudioPlayer src={musicFile} autoPlay id="audioPlayer" controls style={audioStyle} />
                    <h1 className={classes.h1}>Welcome {user.name}</h1>
                    <h2 className={classes.h2}>You have {money} pesos</h2>
                    <input type="submit" onClick={(e) => { changeActivePage(2); }} value={'Fight'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("fighters") }} value={'Fighters'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("bag") }} value={'Bag'} />
                    <input type="submit" onClick={(e) => { changeMultiverseActivePage("shop") }} value={'Shop'} />
                </div>
            }
            {multiverseActivePage === "fighters" && <FightersPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></FightersPage>}
            {multiverseActivePage === "bag" && <ObjectsPage changeMultiverseActivePage={changeMultiverseActivePage} user={user}></ObjectsPage>}
            {multiverseActivePage === "shop" && <ShopPage changeMultiverseActivePage={changeMultiverseActivePage} ></ShopPage>}
        </div>*/
    )
}
export default memo(Multiverse)