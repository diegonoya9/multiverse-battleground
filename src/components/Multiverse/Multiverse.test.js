import React,{startTransition } from 'react';
import { render } from '@testing-library/react';
import { MyContextProvider } from '../../context/MyContext';
import Multiverse from './Multiverse';
// Importa i18next y las funciones necesarias
import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import renderer from 'react-test-renderer';

// Configura i18next
i18n.use(initReactI18next).init({
  lng: 'en', // Configura el idioma deseado
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false,
  },
});
// Mock del contexto para la prueba
const mockUserContext = {
  idUsuario: 1,
  // Otros datos relacionados con el usuario si es necesario
};

test('renders multiverse component', () => {
  // Renderiza Multiverse dentro de MyContextProvider con el contexto simulado
  const component = render(<MyContextProvider value={{ userContext: mockUserContext }}>
    <Multiverse />
  </MyContextProvider>)
  component.findByAltText('mainDiv')

});

it('renders correctly accodring to snapshot', () => {
  const tree = renderer
    .create(
      <MyContextProvider value={{ userContext: mockUserContext }}>
        <React.Suspense fallback={<div>Loading...</div>}>
          {/*
            Envuelve la parte que está causando la suspensión con startTransition
          */}
          {startTransition(() => (
            <Multiverse />
          ))}
        </React.Suspense>
      </MyContextProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});