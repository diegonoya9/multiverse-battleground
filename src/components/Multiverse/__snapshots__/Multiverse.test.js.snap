// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly accodring to snapshot 1`] = `null`;
